parcel_sourcemap_node/src/lib.rs

```diff
@@ -0,0 +1,581 @@
+
+    let vlq_mappings = ctx.get::<JsString>(0)?.into_utf8()?;
+
+    let js_sources_arr = ctx.get::<JsObject>(1)?;
+    let js_sources_arr_len: u32 = js_sources_arr
+        .get_named_property::<JsNumber>("length")?
+        .get_uint32()?;
+    let mut sources = Vec::with_capacity(js_sources_arr_len as usize);
+    for i in 0..js_sources_arr_len {
+        sources.push(
+            js_sources_arr
+                .get_element::<JsString>(i)?
+                .into_utf8()?
+                .into_owned()?,
+        );
+    }
+
+    let js_sources_content_arr = ctx.get::<JsObject>(2)?;
+    let js_sources_content_arr_len: u32 = js_sources_arr
+        .get_named_property::<JsNumber>("length")?
+        .get_uint32()?;
+    let mut sources_content = Vec::with_capacity(js_sources_content_arr_len as usize);
+    for i in 0..js_sources_content_arr_len {
+        sources_content.push(
+            js_sources_content_arr
+                .get_element::<JsString>(i)?
+                .into_utf8()?
+                .into_owned()?,
+        );
+    }
+
+    let js_names_arr = ctx.get::<JsObject>(3)?;
+    let js_names_arr_len: u32 = js_names_arr
+        .get_named_property::<JsNumber>("length")?
+        .get_uint32()?;
+    let mut names = Vec::with_capacity(js_names_arr_len as usize);
+    for i in 0..js_names_arr_len {
+        names.push(
+            js_names_arr
+                .get_element::<JsString>(i)?
+                .into_utf8()?
+                .into_owned()?,
+        );
+    }
+
+    let line_offset = ctx.get::<JsNumber>(4)?.get_int64()?;
+    let column_offset = ctx.get::<JsNumber>(5)?.get_int64()?;
+
+    source_map_instance.add_vlq_map(
+        vlq_mappings.as_slice(),
+        sources.iter().map(|s| &s[..]).collect(),
+        sources_content.iter().map(|s| &s[..]).collect(),
+        names.iter().map(|n| &n[..]).collect(),
+        line_offset,
+        column_offset,
+    )?;
+
+    return ctx.env.get_undefined();
+}
+
+#[js_function]
+fn to_vlq(ctx: CallContext) -> Result<JsObject> {
+    let this: JsObject = ctx.this_unchecked();
+    let source_map_instance: &mut SourceMap = ctx.env.unwrap(&this)?;
+
+    let mut vlq_output: Vec<u8> = vec![];
+    source_map_instance.write_vlq(&mut vlq_output)?;
+    let vlq_string = ctx.env.create_string_from_vec_u8(vlq_output)?;
+    let mut result_obj: JsObject = ctx.env.create_object()?;
+    result_obj.set_named_property("mappings", vlq_string)?;
+    result_obj.set_named_property("sources", _get_sources(&ctx)?)?;
+    result_obj.set_named_property("sourcesContent", _get_sources_content(&ctx)?)?;
+    result_obj.set_named_property("names", _get_names(&ctx)?)?;
+
+    return Ok(result_obj);
+}
+
+#[js_function(1)]
+fn add_indexed_mappings(ctx: CallContext) -> Result<JsUndefined> {
+    let this: JsObject = ctx.this_unchecked();
+    let source_map_instance: &mut SourceMap = ctx.env.unwrap(&this)?;
+
+    // TODO: Figure out a more optimal way of handling typed arrays...
```

Comment:
I believe there is `JsTypedArray`/`JsArrayBuffer`: https://github.com/napi-rs/napi-rs/blob/e4753b7c1d839ab53de42107aeb5395db8749431/napi/src/js_values/arraybuffer.rs.

There's also `JsBuffer` if we wanna use Node buffers. https://napi.rs/concepts/js-values#jsbuffer

Reply:
Yeah been discussing how to properly tackle this with the napi-rs maintainer, gonna update that... it's really hitting performance atm